app/workers/ip_location_worker.rb

```diff
@@ -0,0 +1,13 @@
+class IpLocationWorker
+  include Sidekiq::Worker
+
+  def perform(id, ip_address)
+    @user = User.find(id)
+    location = Pointpin.locate(ip_address)
+    unless location["postcode"] == ("10005" || "11237" || "11385") #locations where we've had offices and therefore triggered a ton of false ip addresses
```

Comment:
Not necessary on every sign in. I was thinking more about the rake task, which will have a ton of users with those ip_addresses, but i can filter them at that point.